src/NUnitFramework/framework/Internal/TestFilter.cs

```diff
@@ -125,6 +125,9 @@ protected virtual bool MatchDescendant(ITest test)
         /// </summary>
         public static TestFilter FromXml(string xmlText)
         {
+            if (string.IsNullOrEmpty(xmlText))
```

Comment:
If you think it's an improvement, which I'm leaning towards, you're already in here anyway...